src/components/PromotedActionsBar.tsx

```diff
@@ -66,8 +70,20 @@ const PromotedActions = {
         icon: Expensicons.Stopwatch,
         text: Localize.translateLocal(`iou.${isTextHold ? 'hold' : 'unhold'}`),
         onSelected: () => {
-            Navigation.dismissModal();
-            ReportUtils.changeMoneyRequestHoldStatus(reportAction);
+            if (!isTextHold) {
+                Navigation.goBack();
+            }
+            const topmostCentralPaneRoute = getTopmostCentralPaneRoute(navigationRef.getRootState() as State<RootStackParamList>);
+            const isReportInRHP = topmostCentralPaneRoute?.name === SCREENS.SEARCH.CENTRAL_PANE;
```

Comment:
NAB: I find the variable name a bit contradictory to the actual code, and it's only used once, so I would get rid of the variable.